Add Forward selection docstring

vcuspinera opened the pull request.

vcuspinera committed:
Add forward selection docstring

techrah commented on feature-selection/forward_select.py:
```diff
@@ -0,0 +1,32 @@
+def forward_select(model, X, y, min_features=None, max_features=10):
```
I see that we are passing `X` and `y` into the function because we don't have a separate `fit` function.

For the Python implementation, I think it would be easier if we followed the scikit-learn template so that our "selectors" fit into the scikit-learn ecosystem easily. That is, you can derive your class from skikit-learn "base classes" and then simply implement `fit` and `transform`.

See the very last example in [Developing scikit-learn estimators](https://scikit-learn.org/stable/developers/develop.html)

That is, instead of implementing a function, we should implement a class.

@jackyho112 @dkruszew What do you guys think?

dkruszew replied:
I agree. Each transformer is a class with a fit and transform function (like we did in 573 lab 2).

An example from sklearn:
https://github.com/scikit-learn/scikit-learn/blob/b194674c42d54b26137a456c510c5fdba1ba23e0/sklearn/feature_selection/_variance_threshold.py

vcuspinera replied:
I did this way because we agree to make only one function for each algorithm, so I was thinking to adapt the [forward algorithm we saw in class](https://github.ubc.ca/MDS-2019-20/DSCI_573_feat-model-select_students/blob/master/solutions/lab2/lab2.ipynb) into one function; however, Ryan's point about being consistent with scikit-learn is very valid.
Jacky, if the four of us agree to have a class for each function that works similar to scikit-learn, probably you should also change the [variance threshold selection](https://github.com/UBC-MDS/feature-selection-python/blob/master/feature-selection/variance-threshold-select.py).
Also we should agree about the general structure of our functions in R, and see if we want to follow similar rules for the 4 functions.

techrah replied:
It's not so clear-cut what to do in R because we are not following any specific framework. Perhaps we can adapt to the caret package later on but we don't have enough experience with this at the moment. I suggest sticking with simple function in R.

techrah replied:
For Python, it's just really easy to do so I don't feel it's a risk.

vcuspinera replied:
Yes, I agree it is a good idea to follow scikit-learn structure, and is not much additional work.

vcuspinera replied:
@ryanhomer, @dkruszew  and @jackyho112  I already make the changes related to the Forward algorithm: from a function to a class, add functions `init`, `fit` and `transform` to the class, and change the name from `forward_select` to `ForwardSelect` as Ryan suggest in slack according to PascalConvention for Python classes

vcuspinera committed:
Update forward selection algorithm

Update the Forward algorithm for Feature Selection, changing it from a function to a class following the scikit-learn structure. Also I change the name of the function from `forward_select` to `ForwardSelect` following the PascalCase convention for naming Python classes.
